core/src/main/resources/org/apache/spark/ui/static/utils.js

```diff
@@ -90,26 +90,27 @@ function formatLogsCells(execLogs, type) {
 }
 
 function getStandAloneAppId(cb) {
-  var words = document.baseURI.split('/');
-  var ind = words.indexOf("proxy");
-  if (ind > 0) {
-    var appId = words[ind + 1];
-    cb(appId);
-    return;
-  }
-  ind = words.indexOf("history");
-  if (ind > 0) {
-    var appId = words[ind + 1];
-    cb(appId);
-    return;
-  }
   // Looks like Web UI is running in standalone mode
   // Let's get application-id using REST End Point
   $.getJSON(uiRoot + "/api/v1/applications", function(response, status, jqXHR) {
     if (response && response.length > 0) {
       var appId = response[0].id;
       cb(appId);
       return;
+    } else {
+      var words = document.baseURI.split('/');
```

Comment:
BTW, have you tested the code changes for Spark UI behind proxy and Spark UI of History server?